lesson10/index.js

```diff
@@ -0,0 +1,70 @@
+const ul = document.getElementById('js-lists');
+
+const listData = [
+    {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+]
+
+function displayLoadingImage () {
+    const loadingImage = document.createElement('img');
+    loadingImage.id = 'loadingImageId';
+    loadingImage.src = './loading-circle.gif';
+    ul.appendChild(loadingImage);
+}
+
+function removeLoadingImage () {
+    const loadingImage = document.getElementById('loadingImageId');
+    loadingImage.remove();
+}
+
+async function getListData() {
+    return new Promise(resolve => {
+        displayLoadingImage ();
+        setTimeout(() => {
+            resolve(listData);
+        }, 3000);
+    });
+}
+
+async function getData() {
+    try {
+        return await getListData();
+    } catch (error) {
+        errorText("エラーが発生しました");
+        return error;
+    } finally {
+        removeLoadingImage ();
+    }
+}
+
+function errorText(text) {
+    const errorText = document.createElement('p');
+    ul.replaceWith(errorText);
+    errorText.textContent = text;
+}
+
+async function createLists() {
+    const frag = new DocumentFragment();
+    const value = await getData();
+    if (value.length) {
+        for(const list of value) {
+            const li = document.createElement('li');
+            const a = document.createElement('a');
+            a.href = list.to;
+            const img = document.createElement('img');
+            img.src = list.img;
+            img.alt = list.alt;
+            const text = document.createTextNode(list.text);
+    
+            a.appendChild(img);
+            a.appendChild(text);
+            li.appendChild(a);
+            frag.appendChild(li);
+        }
+        ul.appendChild(frag);
+    } else {
+        errorText("値が入ってません");
+    }
+}
```

Comment:
There are two jobs in this function
- Determine if the array passed in is empty.
- Creating the list

So, how about this?
```suggestion
async function init() {
  const value = await getData();
  if (value.length) {
    createList(value);
  } else {
    errorText("値が入ってません");
  }
}
init();

const createList = (value) => {
  const frag = new DocumentFragment();
  for (const list of value) {
    const li = document.createElement("li");
    const a = document.createElement("a");
    a.href = list.to;
    const img = document.createElement("img");
    img.src = list.img;
    img.alt = list.alt;
    const text = document.createTextNode(list.text);

    a.appendChild(img);
    a.appendChild(text);
    li.appendChild(a);
    frag.appendChild(li);
  }
  ul.appendChild(frag);
};

```
[codesandbox](https://codesandbox.io/s/serene-swirles-un35t?file=/index.js)

Reply:
Thank you for your reviewing!
I felt it was redundant, so I'm glad I could improve it.

I can write `const createList = (value) => {` or `function createList(value) {`.
Which one is fine?
Are both correct answers?

fixed code -> b0c1021
[codesandbox](https://codesandbox.io/s/bold-thompson-g58yu?file=/index.js)

Reply:
Strictly speaking, they are not the same, but for this function, I don't think it matters which one you use.It is not desirable to mix arrow functions and normal functions without any reason, so let me just tell you that. 

I recommend using the arrow function as it reduces the possibility of unexpected bugs!Try it in your next assignment.

Reply:
I will try using arrow functions in next assignment!!
Thank you so much!!